src/drakvuf.cpp

```diff
@@ -161,6 +161,46 @@ int drakvuf_c::start_plugins(const bool* plugin_list, const plugins_options* opt
     return 1;
 }
 
+int drakvuf_c::stop_plugins(const bool* plugin_list)
+{
+    for (int i = 0; i < __DRAKVUF_PLUGIN_LIST_MAX; i++)
+    {
+        if (plugin_list[i])
+        {
+            int rc = plugins->stop(static_cast<drakvuf_plugin_t>(i));
+            if (rc < 0)
+                return rc;
+        }
+    }
+
+    return 1;
+}
+
+void drakvuf_c::wait_stopped(const bool* plugin_list)
+{
+    auto is_plugins_stopped = [&] (drakvuf_t drakvuf, void*)
+    {
+        if (drakvuf_is_interrupted(drakvuf))
+            return true;
+
+        for (int i = 0; i < __DRAKVUF_PLUGIN_LIST_MAX; i++)
+            if (plugin_list[i] && !plugins->is_stopped(static_cast<drakvuf_plugin_t>(i)))
+                return false;
+
+        return true;
+    };
+
+    auto thunk = +[] (drakvuf_t drakvuf, void* data)
```

Comment:
This is the type of C++ I rather only see inside a plugin ;) What on earth is a `+[]`?

Reply:
@BonusPlay asked me to use lambdas for things like this. And this is a positive lambda: https://stackoverflow.com/a/18889029

I have some hard time to get this too :)

Reply:
For plugins that would be fine. Anything living outside plugins please keep the C++ voodoo to the bare minimum necessary.